internal/system/agent/services.go

```diff
@@ -172,86 +362,90 @@ func getConfig(services []string, ctx context.Context) (ConfigRespMap, error) {
 			if err != nil {
-				m.Metrics[service] = fmt.Sprintf(err.Error())
-				LoggingClient.Error(fmt.Sprintf(err.Error()))
+				out = []byte(fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err.Error()))
+				// m.Metrics[service] = fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err.Error())
+				LoggingClient.Error(fmt.Sprintf(service, err.Error()))
 			} else {
-				LoggingClient.Info(fmt.Sprintf("Registry responded with %s service available", service))
-
-				// Since service is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `service`
-				e, err := registryClient.GetServiceEndpoint(service)
-				if err != nil {
-					m.Metrics[service] = fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err.Error())
-					LoggingClient.Error(fmt.Sprintf(service, err.Error()))
-				} else {
-					// Preparing to add the specified key to the map where the value will be the respective GeneralClient
-					clientInfo := config.ClientInfo{}
-					clientInfo.Protocol = Configuration.Service.Protocol
-					clientInfo.Host = e.Host
-					clientInfo.Port = e.Port
-
-					// This code will evolve to take into account a manifest-like functionality in future. So
-					// rather than assume that the runtime bool flag useRegistry has been initialized to true,
-					// given that the flow has reached this point, having already called functions on the Registry,
-					// such as registryClient.IsServiceAvailable(service), we test for its truthiness. I expect
-					// this code to be refactored as we evolve toward a manifest-like functionality in future.
-					usingRegistry := false
-					if registryClient != nil {
-						usingRegistry = true
-					}
-
-					Configuration.Clients[e.ServiceId] = clientInfo
-					params := types.EndpointParams{
-						ServiceKey:  e.ServiceId,
-						Path:        "/",
-						UseRegistry: usingRegistry,
-						Url:         Configuration.Clients[e.ServiceId].Url() + clients.ApiMetricsRoute,
-						Interval:    internal.ClientMonitorDefault,
-					}
-					// TODO: The following note is related to future work:
-					// TODO: With the current deployment strategy, GeneralClient's func init(params types.EndpointParams) {...} returns blank "url"...
-					// TODO: [Need a manifest-like functionality in future...]
-					// Add the service key to the map where the value is the respective GeneralClient
-					generalClients[e.ServiceId] = general.NewGeneralClient(params, startup.Endpoint{RegistryClient: &registryClient})
+				// Preparing to add the specified key to the map where the value will be the respective GeneralClient
+				clientInfo := config.ClientInfo{}
+				clientInfo.Protocol = Configuration.Service.Protocol
+				clientInfo.Host = e.Host
+				clientInfo.Port = e.Port
+
+				// This code will evolve to take into account a manifest-like functionality in future. So
```

Comment:
I'm not sure I understand this comment - is this saying that if the registerClient is available then use the registry, otherwise don't?

Also I think large comments like this are prone to growing stale in that when the manifest functionality is eventually implemented, it's unfortunately often the case that updating/removing comments like this is forgotten and since it's already in the codebase when that functionality is implemented, reviewers may not know to look at this part of the code to see that it should be updated, so I'd suggest removing this. I think a more appropriate comment is  TODO: as that's easier for reviewers to see when the feature is implemented that the TODO should be removed